__tests__/app.test.js

```diff
@@ -15,28 +16,40 @@ beforeEach(() => seed(data));
 afterAll(() => db.end());
 
 describe("/api", () => {
-	describe("GET: 200 /topics", () => {
-		test("should respond with status code 200", () => {
-			return request(app).get("/api/topics").expect(200);
-		});
-		test("should respond with an array of topic objects, each of which should have the following properties: slug, description", () => {
-			return request(app)
-				.get("/api/topics")
-				.expect(200)
-				.then((response) => {
-					const { topics } = response.body;
-					expect(Array.isArray(topics)).toBe(true);
-					topics.forEach((topic) => {
-						expect(topic).toHaveProperty(
-							"slug",
-							expect.any(String)
-						);
-						expect(topic).toHaveProperty(
-							"description",
-							expect.any(String)
-						);
+	test("GET: 200 should respond with an object object describing all the available endpoints on this API", () => {
+		fs.readFile("./endpoints.json", "utf8")
+			.then((fileContents) => {
+				return JSON.parse(fileContents);
+			})
+			.then((expected) => {
+				return request(app)
+					.get("/api")
+					.expect(200)
+					.then((response) => {
+						const { endpoints } = response.body;
+						expect(endpoints).toMatchObject(expected);
 					});
+			});
+	});
+});
+describe("/topics", () => {
+	test("GET: 200 should respond with status code 200", () => {
+		return request(app).get("/api/topics").expect(200);
+	});
+	test("GET: should respond with an array of topic objects, each of which should have the following properties: slug, description", () => {
+		return request(app)
+			.get("/api/topics")
+			.expect(200)
+			.then((response) => {
+				const { topics } = response.body;
+				expect(Array.isArray(topics)).toBe(true);
+				topics.forEach((topic) => {
+					expect(topic).toHaveProperty("slug", expect.any(String));
+					expect(topic).toHaveProperty(
+						"description",
+						expect.any(String)
+					);
 				});
-		});
+			});
 	});
```

Comment:
Please add a test for a potentially that may occur when you got to a path that does not exist

Reply:
added `app.all("/api/*"...` on app.js